src/main/java/jenkins/plugins/http_request/auth/BasicDigestAuthentication.java

```diff
@@ -1,82 +0,0 @@
-package jenkins.plugins.http_request.auth;
-
-import java.io.PrintStream;
-
-import org.apache.http.client.methods.HttpRequestBase;
-import org.apache.http.client.utils.URIUtils;
-import org.apache.http.impl.client.CloseableHttpClient;
-import org.apache.http.impl.client.HttpClientBuilder;
-import org.apache.http.protocol.HttpContext;
-import org.kohsuke.stapler.DataBoundConstructor;
-import org.kohsuke.stapler.QueryParameter;
-
-import edu.umd.cs.findbugs.annotations.NonNull;
-import hudson.Extension;
-import hudson.model.AbstractDescribableImpl;
-import hudson.model.Descriptor;
-import hudson.util.FormValidation;
-
-import jenkins.plugins.http_request.HttpRequestGlobalConfig;
-
-/**
- * @author Janario Oliveira
- * @deprecated use Jenkins credentials, marked to remove in 1.8.19
- */
-@Deprecated
-public class BasicDigestAuthentication extends AbstractDescribableImpl<BasicDigestAuthentication>
-        implements Authenticator {
-	private static final long serialVersionUID = 4818288270720177069L;
-
-	private final String keyName;
-    private final String userName;
-    private final String password;
```

Comment:
FTR the only required change was to change this field's type to `hudson.util.Secret` (ideally also constructor parameter and getter return value), and adapt related code as needed.

Reply:
Thanks for the review! 

The class is marked as deprecated for years  and the UI indicates that too. I wouldn't consider this worth the effort of an API change / migration. It's a good chance to remove it for good now. There shouldn't be any use case left … hopefully.

I don't have a strong opinion on this though. So if either making fields transient or migration to Secret are more suitable solution to fix this I can update the PR.

Reply:
Is there any workflow on how to remove a class?

If I understand this correctly transient (all fields?) + UI removal is the preferred way; PR update is on the way.

Reply:
> Is there any workflow on how to remove a class?

Good question, it's not a case I had do handle before -- usually old types are kept around to do migration of old configuration or data to the new types, and then `transient` + `readResolve` do the job.

I'd probably keep the class, and make the field holding it `transient`. Then the idea would be to make the class an empty shell. I'd try implementing `#readObject(ObjectInputStream)` or an X-Stream `Converter` (as a [nested class called `ConverterImpl`](https://github.com/jenkinsci/jenkins/blob/082634cf1e4ef91cac8c82815e52046083722e36/core/src/main/java/hudson/util/XStream2.java#L425-L429)) to bypass any warnings from default behaviors, and basically do as little as possible.

(You could customize the deserialization of the class holding the field, but that doesn't seem like a great idea for longer-term maintainability.)

---

Regarding the comment that started this thread, it was only to point out what a minimal fix would have looked like. We're generally not asking maintainers to rip out entire features, and I did not want this PR to give this impression.